torch/csrc/api/include/torch/nn/options/activation.h

```diff
@@ -315,5 +315,102 @@ struct TORCH_API GumbelSoftmaxFuncOptions {
 
 } // namespace functional
 
+// ============================================================================
+
+/// Options for MultiheadAttention functional and module.
+struct TORCH_API MultiheadAttentionOptions {
+  MultiheadAttentionOptions(int64_t embed_dim, int64_t num_heads)
+    : embed_dim_(embed_dim), num_heads_(num_heads) {
+    if (!kdim_) {
+      kdim_ = embed_dim;
+    }
+    if (!vdim_) {
+      vdim_ = embed_dim;
+    }
+  }
+
+  /// total dimension of the model.
+  TORCH_ARG(int64_t, embed_dim);
+
+  /// parallel attention heads.
+  TORCH_ARG(int64_t, num_heads);
+
+  /// a Dropout layer on attn_output_weights. Default: 0.0.
+  TORCH_ARG(double, dropout) = 0.0;
+
+  /// add bias as module parameter. Default: true.
+  TORCH_ARG(bool, bias) = true;
+
+  /// add bias to the key and value sequences at dim=0.
+  TORCH_ARG(bool, add_bias_kv) = false;
+
+  /// add a new batch of zeros to the key and value sequences at dim=1.
+  TORCH_ARG(bool, add_zero_attn) = false;
+
+  /// total number of features in key. Default: c10::nullopt.
+  TORCH_ARG(c10::optional<int64_t>, kdim) = c10::nullopt;
+
+  /// total number of features in key. Default: c10::nullopt.
+  TORCH_ARG(c10::optional<int64_t>, vdim) = c10::nullopt;
+};
+
+// ============================================================================
+
+namespace functional {
+
+/// Options for `torch::nn::functional::multi_head_attention_forward`
+struct TORCH_API MultiheadAttentionForwardOptions {
```

Comment:
We might want to change it to `MultiheadAttentionForwardFuncOptions`, to be more consistent with the naming scheme of other functional options :D